backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala

```diff
@@ -1,54 +1,55 @@
 package com.softwaremill.bootzooka.email
 
-import com.softwaremill.bootzooka.infrastructure.Doobie._
-import monix.eval.{Fiber, Task}
+import cats.Parallel
+import cats.effect.{Fiber, IO}
 import com.softwaremill.bootzooka.email.sender.EmailSender
+import com.softwaremill.bootzooka.infrastructure.Doobie._
 import com.softwaremill.bootzooka.metrics.Metrics
 import com.softwaremill.bootzooka.util.IdGenerator
 import com.typesafe.scalalogging.StrictLogging
-import monix.execution.Scheduler.Implicits.global
 
-/** Schedules emails to be sent asynchronously, in the background, as well as manages sending of emails in batches.
-  */
-class EmailService(emailModel: EmailModel, idGenerator: IdGenerator, emailSender: EmailSender, config: EmailConfig, xa: Transactor[Task])
+/** Schedules emails to be sent asynchronously, in the background, as well as manages sending of emails in batches. */
+class EmailService(emailModel: EmailModel, idGenerator: IdGenerator, emailSender: EmailSender, config: EmailConfig, xa: Transactor[IO])
     extends EmailScheduler
     with StrictLogging {
 
   def apply(data: EmailData): ConnectionIO[Unit] = {
     logger.debug(s"Scheduling email to be sent to: ${data.recipient}")
-    idGenerator.nextId().to[ConnectionIO].flatMap(id => emailModel.insert(Email(id, data)))
+    idGenerator
+      .nextId[ConnectionIO, Email]()
+      .flatMap(id => emailModel.insert(Email(id, data)))
   }
 
-  def sendBatch(): Task[Unit] = {
+  def sendBatch(): IO[Unit] = {
     for {
       emails <- emailModel.find(config.batchSize).transact(xa)
       _ = if (emails.nonEmpty) logger.info(s"Sending ${emails.size} emails")
-      _ <- Task.sequence(emails.map(_.data).map(emailSender.apply))
+      _ <- IO(emails.map(_.data).map(emailSender.apply))
```

Comment:
what happened to .sequence? This won't work, we're constructing an `IO[IO[Unit]]` and ignoring the inner result